### ISSUE TITLE
Rule proposal: `no-untyped-empty-array` in ts file

### ISSUE THREAD

**Issue by zanminkian** [sindresorhus/eslint-plugin-unicorn#2396]
### Description

Empty array in ts is not type safe.
```ts
const arr = []; // type of arr is `any[]`
```

### Fail

```ts
const arr = [];
```

### Pass

```ts
const arr: number[] = [];
const arr: any[] = [];
const arr = [] as unknow[];
const arr = ['foo'];
```

### Proposed rule name

no-untyped-empty-array

**Comment by zanminkian** [comment]
Maybe `@typescript-eslint/eslint-plugin` should implement this rule. 🤔️

**Comment by sindresorhus** [comment]
This would indeed be a better fit there.
